api/src/schema/resolvers.js

```diff
@@ -50,30 +42,21 @@ const resolvers = {
       }
 
       if (filter) {
-        for (let key in filter) {
-          // make sure the key exists in the comparators' keys
-          if (Object.keys(comparators).includes(key)) {
-            // if the value is a number
-            // - optionally starting with a minus sign
-            // - containing zero or one decimal places
-            // convert it to a float instead of a string
-            if (filter[key].match(/^-?\d+\.?\d+$/)) {
-              filter[key] = parseFloat(filter[key])
-            }
-
-            // We are evaling our own code here, not user input.
-            // Filter.field is our code, stringified and stored in an enum.
-            // The user choses one of the enum values and we convert the string
-            // back to a function which accepts a matcher and generates the
-            // query we need to find that field.
-            // eslint-disable-next-line no-eval
-            let queryGenerator = eval(filter.field)
-            let attrQuery = queryGenerator({
-              [comparators[key]]: filter[key],
-            })
-            query['$and'].push(attrQuery)
-          }
+        let value = filter.value
+        if (filter.value.match(/^-?\d+\.?\d+$/)) {
+          value = parseFloat(filter.value)
```

Comment:
I think it's worth keeping in the comment describing the regex, in one form or another.